backend/src/test/java/net/vidux/camhub/discovery/CameraScanTest.java

```diff
@@ -0,0 +1,61 @@
+package net.vidux.camhub.discovery;
+
+import java.util.Set;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class CameraScanTest {
+
+  @Test
+  void testViduxHelper() {
+    Assertions.assertDoesNotThrow(
+        () -> {
+          CameraScan viduxHelper = new CameraScanViduxHelper();
+          Set<RawCameraData> set = viduxHelper.scanCams().get();
+
+          for (RawCameraData cam : set) {
+            Assertions.assertNotNull(cam.getName());
+            Assertions.assertNotNull(cam.getFirmware());
+            Assertions.assertNotNull(cam.getIpAddress());
+            Assertions.assertNotNull(cam.getSerialNumber());
+            Assertions.assertEquals(9, cam.getSerialNumber().length());
+          }
```

Comment:
Amikor ellenőrizzük a metódus által visszaadott értékeket, akkor arra kell törekedni, hogy minél egzaktabb módon vizsgáljuk az értékeket.
Ez azt jelenti, hogy ahol lehet, ott nem csak azt ellenőrizzük le, hogy nem null-e az érték, hanem a konkrét értéket is összevetjük azzal, amit elvárunk.

Ezt egyszerűen meg tudod tenni, ha csinálsz egy elvárt értékeket tartalmazó Set-et, amit összehasonlítasz assertEquals-al az eredményül kapott Set-el.